shared/src/commonMain/kotlin/org/hyperskill/app/progress_screen/presentation/ProgressScreenActionDispatcher.kt

```diff
@@ -50,59 +51,39 @@ internal class ProgressScreenActionDispatcher(
-                onNewMessage(ProgressScreenFeature.TrackWithProgressFetchResult.Error)
-                return@coroutineScope
-            }
-            val studyPlan = studyPlanDeferred.await().getOrElse {
-                sentryInteractor.finishTransaction(transaction, throwable = it)
-                onNewMessage(ProgressScreenFeature.TrackWithProgressFetchResult.Error)
-                return@coroutineScope
-            }
+        sentryInteractor.withTransaction(
+            HyperskillSentryTransactionBuilder.buildProgressScreenRemoteTrackWithProgressLoading(),
+            onError = { ProgressScreenFeature.TrackWithProgressFetchResult.Error }
+        ) {
+            coroutineScope {
+                val profileDeferred = async {
+                    currentProfileStateRepository
+                        .getState(forceUpdate = action.forceLoadFromNetwork)
+                }
+                val studyPlanDeferred = async {
+                    currentStudyPlanStateRepository
+                        .getState(forceUpdate = action.forceLoadFromNetwork)
+                }
 
-            if (studyPlan.trackId == null) {
-                sentryInteractor.finishTransaction(transaction)
-                onNewMessage(ProgressScreenFeature.TrackWithProgressFetchResult.Error)
-                return@coroutineScope
-            }
+                val profile = profileDeferred.await().getOrThrow()
+                val studyPlan = studyPlanDeferred.await().getOrThrow()
 
-            val trackWithProgress = fetchTrackWithProgress(studyPlan.trackId, action.forceLoadFromNetwork)
-                .getOrElse {
-                    sentryInteractor.finishTransaction(transaction, throwable = it)
-                    onNewMessage(ProgressScreenFeature.TrackWithProgressFetchResult.Error)
-                    return@coroutineScope
+                if (studyPlan.trackId == null) {
+                    return@coroutineScope ProgressScreenFeature.TrackWithProgressFetchResult.Error
                 }
 
-            if (trackWithProgress == null) {
-                sentryInteractor.finishTransaction(transaction)
-                onNewMessage(ProgressScreenFeature.TrackWithProgressFetchResult.Error)
-                return@coroutineScope
-            }
+                val trackWithProgress =
+                    fetchTrackWithProgress(studyPlan.trackId, action.forceLoadFromNetwork)
+                        .getOrThrow()
+                        ?: return@coroutineScope ProgressScreenFeature.TrackWithProgressFetchResult.Error
 
-            sentryInteractor.finishTransaction(transaction)
-            onNewMessage(
                 ProgressScreenFeature.TrackWithProgressFetchResult.Success(
                     trackWithProgress = trackWithProgress,
                     studyPlan = studyPlan,
                     profile = profile
                 )
-            )
-        }
+            }
+        }.let(onNewMessage)
```

Comment:
Just a minor question:

Personally for me, the use of `let` means that we want to execute a code block containing a non-null value with a safe call operator `?` call before.

What's preferable to use here `let` or `also` scope function?

Reply:
In this case, there isn't a big difference in which scope function to use.
But, in my opinion, `let` is preferable according to the name of the function.